unit_test/fuzzing/test_requester/test_spdm_requester_negotiate_algorithms/negotiate_algorithms.c

```diff
@@ -161,6 +165,98 @@ void libspdm_test_requester_negotiate_algorithms_case4(void **State)
+    spdm_context->local_context.algorithm.key_schedule = m_libspdm_use_key_schedule_algo;
+    spdm_context->local_context.algorithm.other_params_support = 0;
+
+    spdm_context->local_context.capability.flags = SPDM_GET_CAPABILITIES_REQUEST_FLAGS_KEY_EX_CAP |
+                                                   SPDM_GET_CAPABILITIES_REQUEST_FLAGS_MUT_AUTH_CAP|
+                                                   SPDM_GET_CAPABILITIES_REQUEST_FLAGS_ENCRYPT_CAP|
+                                                   SPDM_GET_CAPABILITIES_REQUEST_FLAGS_MAC_CAP|
+                                                   SPDM_GET_CAPABILITIES_REQUEST_FLAGS_PSK_CAP;
+    connection_capability_flags =
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_KEY_EX_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_ENCRYPT_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MAC_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MUT_AUTH_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEAS_CAP |
+        SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_PSK_CAP;
+
+    spdm_context->connection_info.capability.flags = connection_capability_flags;
+
+    /* Sub Case 0: Initially*/
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 1: MEL_CAP set 1,mel_specification set SPDM_MEL_SPECIFICATION_DMTF*/
+    spdm_context->connection_info.capability.flags |= SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEL_CAP;
+    libspdm_reset_message_a(spdm_context);
+    spdm_context->local_context.algorithm.mel_spec = SPDM_MEL_SPECIFICATION_DMTF;
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 2: MEL_CAP set 1, mel_specification set 0*/
+    spdm_context->connection_info.capability.flags |= SPDM_GET_CAPABILITIES_RESPONSE_FLAGS_MEL_CAP;
+    libspdm_reset_message_a(spdm_context);
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    spdm_context->local_context.algorithm.mel_spec = 0;
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 3:MEL_CAP set 0, mel_specification set SPDM_MEL_SPECIFICATION_DMTF*/
+    spdm_context->connection_info.capability.flags = connection_capability_flags;
+    libspdm_reset_message_a(spdm_context);
+    spdm_context->local_context.algorithm.mel_spec = SPDM_MEL_SPECIFICATION_DMTF;
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 4: MEL_CAP set 0, mel_specification set 0*/
+    spdm_context->connection_info.capability.flags = connection_capability_flags;
+    libspdm_reset_message_a(spdm_context);
+    spdm_context->connection_info.connection_state = LIBSPDM_CONNECTION_STATE_AFTER_CAPABILITIES;
+    spdm_context->local_context.algorithm.mel_spec = 0;
+    libspdm_negotiate_algorithms(spdm_context);
+
+    /* Sub Case 5: MULTI_KEY_CAP set 01*/
```

Comment:
Change
`/* Sub Case 5: MULTI_KEY_CAP set 01*/`
To
`/* Sub Case 5: MULTI_KEY_CAP set 01b*/`